### PR TITLE
Add footer, fix some UI bugs

### PR DESCRIPTION
**Pull request by Michal-Ston-Mobica**
Footer with info added. 
The footer contains info about the project, known issues and OSS technologies used (until we write a proper README).
Some of the buttons logic was improved, eg. start button (now "Load") will be disabled when the file is being processed or there are no files on the list.

### CODE REVIEW

File: src/wasm/html/index.html [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <head>
     <meta charset="utf-8">
     <meta name="viewport" content="width=device-width, initial-scale=1">
-    <title>innoextract wasm</title>
+    <title>Innoextract wasm</title>
```
Comment (Rafal-Gorczewski-Mobica) [inline src/wasm/html/index.html]:
```suggestion
    <title>Innoextract WASM</title>
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -594,6 +593,7 @@ if(WASM)
 	file(COPY src/wasm/html/css DESTINATION "${CMAKE_CURRENT_BINARY_DIR}")
 	file(COPY src/wasm/html/img DESTINATION "${CMAKE_CURRENT_BINARY_DIR}")
 	string(TIMESTAMP BUILD_DATE "%s")
+    execute_process(COMMAND bash -c "git describe --dirty --always --tags" OUTPUT_VARIABLE GIT_VERSION)
```
Comment (Rafal-Gorczewski-Mobica) [inline CMakeLists.txt]: What if someone doesn't have `bash`?

File: src/wasm/html/index.html [hunk 3]
```diff
@@ -14,98 +14,131 @@
...
-              <h5 class="modal-title"><i class="bi bi-exclamation-circle-fill"></i> Error</h5>
-              <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
+        <div id="footer" class="col-xl-8 mx-auto p-xl-5 p-3 fixed-bottom">
+          <hr>
+        <small>
+          <a class="collapse-name" onClick='unCollapse("about", this)' role="button">About</a>
+          • <a class="collapse-name" onClick='unCollapse("issues", this)' role="button">Known issues</a>
+          • <a class="collapse-name" onClick='unCollapse("techs", this)' role="button">Open source technologies</a>
+          • <a class="collapse-name" href="https://github.com/Mobica/innoextract-wasm" target="_blank">Github repository</a>
+          <div class="float-end">Innoextract-wasm ${GIT_VERSION}</div>
```
Comment (Rafal-Gorczewski-Mobica) [inline src/wasm/html/index.html]:
```suggestion
          <div class="float-end">innoextract-wasm ${GIT_VERSION}</div>
```

File: src/wasm/html/index.html [hunk 4]
```diff
@@ -14,98 +14,131 @@
...
+          <div id="footer-group">
+            <div id="about" class="collapsible collapsed">
+                <p><a href="https://github.com/Mobica/innoextract-wasm">Innoextract-wasm</a> is created as a side project by <a href="https://mobica.com/">Mobica</a> employees. The main goal was to learn a bit of WASM and porting C/C++ code to a web application using emscripten.<br>
```
Comment (Rafal-Gorczewski-Mobica) [inline src/wasm/html/index.html]:
```suggestion
                <p><a href="https://github.com/Mobica/innoextract-wasm">innoextract-wasm</a> is created as a side project by <a href="https://mobica.com/">Mobica</a> employees. The main goal was to learn a bit of WASM and porting C++ code to a web application using Emscripten.<br>
```

File: src/wasm/html/index.html [hunk 5]
```diff
@@ -14,98 +14,131 @@
...
+                  The <a href="https://constexpr.org/innoextract/">original Innoextract</a> project was created by <a href="https://ds.me/">Daniel Scharrer</a>. This port is not directly related to Daniel's work, so any bugs or feature requests regarding this website should not be sent to him.</p>
             </div>
-            <div class="modal-body">
-              <p>Innoextract has failed!</p>
-              <p id="errorMsg"></p>
+            <div id="issues" class="collapsible collapsed">
+                <ul>
+                <li>Some installers are using seek() on output file, which is not supported at this moment (eg. Stronghold HD)</li>
```
Comment (Rafal-Gorczewski-Mobica) [inline src/wasm/html/index.html]:
```suggestion
                <li>Some installers use seek() on the output file, which is not supported at this moment (eg. Stronghold HD)</li>
```

File: src/wasm/html/js/innoextract.misc.js [hunk 6]
```diff
@@ -301,3 +303,28 @@ function abortExtraction() {
         console.debug("Abort requested");
     });
 }
+
+function unCollapse(id, button) {
```
Comment (Rafal-Gorczewski-Mobica) [inline src/wasm/html/js/innoextract.misc.js]:
```suggestion
function uncollapse(id, button) {
```

File: src/wasm/html/index.html [hunk 7]
```diff
@@ -14,98 +14,131 @@
...
-    <div id="errorModal" class="modal" tabindex="-1">
-        <div class="modal-dialog">
-          <div class="modal-content">
-            <div class="modal-header">
-              <h5 class="modal-title"><i class="bi bi-exclamation-circle-fill"></i> Error</h5>
-              <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
+        <div id="footer" class="col-xl-8 mx-auto p-xl-5 p-3 fixed-bottom">
+          <hr>
+        <small>
+          <a class="collapse-name" onClick='unCollapse("about", this)' role="button">About</a>
```
Comment (Rafal-Gorczewski-Mobica) [inline src/wasm/html/index.html]:
```suggestion
          <a class="collapse-name" onClick='uncollapse("about", this)' role="button">About</a>
```

File: src/wasm/html/index.html [hunk 8]
```diff
@@ -14,98 +14,131 @@
...
+          • <a class="collapse-name" onClick='unCollapse("issues", this)' role="button">Known issues</a>
```
Comment (Rafal-Gorczewski-Mobica) [inline src/wasm/html/index.html]:
```suggestion
          • <a class="collapse-name" onClick='uncollapse("issues", this)' role="button">Known issues</a>
```

File: src/wasm/html/index.html [hunk 9]
```diff
@@ -14,98 +14,131 @@
...
+          • <a class="collapse-name" onClick='unCollapse("techs", this)' role="button">Open source technologies</a>
```
Comment (Rafal-Gorczewski-Mobica) [inline src/wasm/html/index.html]:
```suggestion
          • <a class="collapse-name" onClick='uncollapse("techs", this)' role="button">Open source technologies</a>
```

**Review by Rafal-Gorczewski-Mobica — CHANGES_REQUESTED** [review]

File: src/wasm/html/index.html (hunk 3 above)
Comment (Michal-Ston-Mobica) [reply src/wasm/html/index.html]: I'll keep the capitalization as is, with first letter being upper-case.

File: src/wasm/html/index.html (hunk 4 above)
Comment (Michal-Ston-Mobica) [reply src/wasm/html/index.html]: Innoextract as-is, capitalized Emscripten.

**Review by Rafal-Gorczewski-Mobica — APPROVED** [review]

File: src/wasm/.clang-format
Comment (Rafal-Gorczewski-Mobica) [inline src/wasm/.clang-format:1]: Something's wrong. Why was this file removed?

Comment (Michal-Ston-Mobica) [reply src/wasm/.clang-format:1]: It's probably because I've ended up merging the branches, and this file got lost in the process of making the commit.
Re-added now.
